src/databases/indexedDb/indexedDbOps.ts

```diff
@@ -38,14 +41,17 @@ async function deleteRecords() {
 
 	if (!selectedTable) return;
 
-	const primaryKey = Array.isArray(selectedTable.schema.primKey.keyPath)
-		? (selectedTable.schema.primKey.keyPath[0] as string)
-		: (selectedTable.schema.primKey.keyPath as string);
+	const primKeyPath = selectedTable.schema.primKey.keyPath;
+	const primaryKey = Array.isArray(primKeyPath)
+		? (primKeyPath[0] as string)
+		: (primKeyPath as string);
```

Comment:
Please remove the type assertion here, as `primaryKey` can legitimately be either a `string` or `undefined`.

```suggestion
	const primaryKey = Array.isArray(primKeyPath)
		? (primKeyPath[0])
		: (primKeyPath);
```